### ISSUE TITLE
CI fails for 3.0.x and 2.2.x due to recent pytest version

### ISSUE THREAD

**Issue by timhoffm** [matplotlib/matplotlib#12825]
### Bug report

CI fails on branches 3.0.x and 2.2.x fails with

~~~
E               _pytest.warning_types.RemovedInPytest4Warning: MarkInfo objects are deprecated as they contain merged marks which are hard to deal with correctly.
E               Please use node.get_closest_marker(name) or node.iter_markers(name).
E               Docs: https://docs.pytest.org/en/latest/mark.html#updating-code
~~~

See e.g. #12811, #12812.

This is due to a recent pytest version.

Solution: Either pin pytest<3.8 (I think that's the correct number) or backport #12154 (and maybe  #12297).

**Comment by QuLogic** [comment]
Note also that backporting those would bumpy the minimum required version as well.

**Comment by timhoffm** [comment]
Fixed by #12878 and #12879.
